client/my-sites/plugins/marketplace/controller.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal Dependencies
+ */
+import { getSiteFragment } from 'calypso/lib/route';
+import AsyncLoad from 'calypso/components/async-load';
+import MarketplacePluginDetails from 'calypso/my-sites/plugins/marketplace/marketplace-plugin-details';
+
+export function renderMarketplacePlugin( context, next ) {
+	const siteUrl = getSiteFragment( context.path );
+
+	const pluginSlug = decodeURIComponent( context.params.plugin );
+
+	context.primary = (
+		<MarketplacePluginDetails
+			path={ context.path }
+			marketplacePluginSlug={ pluginSlug }
+			siteUrl={ siteUrl }
+		/>
+	);
+
+	next();
+}
+
+export function renderDomainsPage( context, next ) {
+	context.primary = (
+		<AsyncLoad require="calypso/my-sites/plugins/marketplace/marketplace-domain-upsell" />
+	);
+	next();
+}
+
+export function renderPluginsSetupStatusPage( context, next ) {
+	context.primary = (
+		<AsyncLoad require="calypso/my-sites/plugins/marketplace/marketplace-plugin-setup-status" />
+	);
+	next();
+}
+
+export function renderMarketplaceThankYou( context, next ) {
+	// context.primary = (
+	// 	<AsyncLoad require="calypso/my-sites/plugins/marketplace/marketplace-stand-alone-thank-you" />
+	// );
+	context.primary = <div>Not Implemented</div>;
```

Comment:
I guess let's add some `@TODO` comments here as well. They often further help express intention.

Reply:
These will be addressed by my immediate next commit that is why I did not put anything extra :)

Reply:
Sounds good. Feel free to split to multiple PRs if that helps 😉